components/common/infinite_scroll.jsx

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const SCROLL_BUFFER = 10;
+
+export default class InfiniteScroll extends React.PureComponent {
+    static defaultProps = {
+        bufferValue: SCROLL_BUFFER,
+    };
+
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+
+        /**
+         * Function that is called to load more items
+         */
+        callBack: PropTypes.func.isRequired,
+
+        /**
+         * Boolean to indicate that no more data will be loaded
+         */
+        endOfData: PropTypes.bool.isRequired,
+
+        /**
+         * Message to display when all the data has been scrolled through
+         */
+        endOfDataMessage: PropTypes.string.isRequired,
+
+        /**
+         * A wrapper class to define styling of the infinite scroll
+         */
+        styleClass: PropTypes.string.isRequired,
+
+        /**
+         * A number that determines how far the scroll is near the bottom before
+         * loading more items
+         */
+        bufferValue: PropTypes.number,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isFetching: false,
+        };
+        this.node = React.createRef();
+    }
+
+    componentDidMount() {
+        this.node.current.addEventListener('scroll', this.handleScroll.bind(this));
```

Comment:
We don't need to bind `this` here as handle scroll is an arrow function it gets the context of this from the class

Reply:
Oh yeah... that's true. Let me make that change.

Reply:
@sudheerDev I have finished making this change